python/tvm/micro/testing/pytest_plugin.py

```diff
@@ -0,0 +1,111 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=invalid-name,redefined-outer-name
+""" microTVM testing fixtures used to deduce testing argument
+    values from testing parameters """
+
+import pathlib
+import os
+import datetime
+import pytest
+
+from tvm.contrib.utils import tempdir
+
+from .utils import get_supported_boards
+
+
+def pytest_addoption(parser):
+    """Adds more pytest arguments"""
+    parser.addoption(
+        "--board",
```

Comment:
maybe we should namespace this since it could be shown in the test `--help` for pytests outside of microtvm. `--microtvm-board`? same with the ones below.

Reply:
Do you mean it would show up in other tests in TVM? I don't think so since you have to include it specifically like this in the conftest.py:
```
pytest_plugins = [
    "tvm.micro.testing.pytest_plugin",
]
```

Reply:
oh i see, but what if you ran `pytest tests/micro` and ran both Arduino and Zephyr tests? or what if we refactored this later?

Reply:
I tested it, if you run `pytest tests/micro` it doesn't recognize `--board`. If we refactor it at the TVM level, yeah we should change it then. For now, I prefer `--board` to keep is simple. thoughts?

Reply:
ok, fine for now.